LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabClient.java

```diff
@@ -0,0 +1,185 @@
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+import com.microsoft.identity.labapi.utilities.exception.LabError;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
+public class LabClient implements ILabClient {
+
+    private final LabApiAuthenticationClient mLabApiAuthenticationClient;
+
+    /**
+     * Temp users API provided by Lab team can often take more than 10 seconds to return...hence, we
+     * are overriding the read timeout.
+     */
+    private static final int TEMP_USER_API_READ_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(15);
+
+    @Override
+    public LabAccount fetchUser(@NonNull final LabQuery labQuery) throws LabApiException {
+        final List<ConfigInfo> configInfos = fetchConfigsFromLab(labQuery);
+        // for each query, lab actually returns a list of accounts..all of which fit the criteria..
+        // usually we only need one such account, and hence over here we are just picking the first
+        // element of the list.
+        final ConfigInfo configInfo = configInfos.get(0);
+        return createAccount(configInfo);
+    }
+
+    @Override
+    public List<LabAccount> fetchUsers(@NonNull final LabQuery labQuery) throws LabApiException {
+        final List<ConfigInfo> configInfos = fetchConfigsFromLab(labQuery);
+
+        final List<LabAccount> labAccounts = new ArrayList<>(configInfos.size());
+
+        for (final ConfigInfo configInfo : configInfos) {
+            labAccounts.add(createAccount(configInfo));
+        }
+
+        return labAccounts;
+    }
+
+    private LabAccount createAccount(@NonNull final ConfigInfo configInfo) throws LabApiException {
+        // for guest accounts the UPN is located under homeUpn field
+        String username = configInfo.getUserInfo().getHomeUPN();
+        if (username == null || username.equals("") || username.equalsIgnoreCase("None")) {
+            // for accounts that are NOT guest..the UPN is directly on the UPN field
+            username = configInfo.getUserInfo().getUpn();
+        }
+
+        final String password = getPassword(configInfo);
+
+        return new LabAccount(username, password, UserType.fromName(
+                configInfo.getUserInfo().getUserType())
+        );
+    }
+
+    private List<ConfigInfo> fetchConfigsFromLab(@NonNull final LabQuery query) throws LabApiException {
+        Configuration.getDefaultApiClient().setAccessToken(
+                mLabApiAuthenticationClient.getAccessToken()
+        );
+        try {
+            final ConfigApi api = new ConfigApi();
+            return api.apiConfigGet(
+                    query.getUserType() != null ? query.getUserType().getValue() : null,
```

Comment:
>query.getUserType() != null ? query.getUserType().getValue() : null, [](http://example.com/codeflow?start=20&length=68)

if you override the toString() method in the enums this line will simply become.. "query.getUserType()".
See my comment in AltIdSource.java

Reply:
hmm..actually it wouldn't work because the `apiConfigGet` method wants `Strings`...but `query.getUserType()` would return something of type `UserType` and the method wouldn't accept it:

![image](https://user-images.githubusercontent.com/37125644/121419502-dedb9100-c920-11eb-8bdd-56ab987f302c.png)

Reply:
It would work if you were doing string manipulation, but that's because the string + string gets turned into new StringBuilder.append(string).append(string).toString() by the JVM.  There's also More Magic(tm) with strings.  I bet it would if you did "" + foo, as the argument as well.

Reply:
One way to make it work would be to use String.valueOf()...
https://www.jdoodle.com/ia/ex2

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1406/commits/d5e4b3943a0944faaab79d0fecb91d175da48c72 (using a custom valueOf method...because String.valueOf wouldn't work for our purposes)